react/Paywall/OnlyOfficePaywall.jsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react'
+import ReactMarkdown from 'react-markdown'
+
+import { useClient } from 'cozy-client'
+import {
+  arePremiumLinksEnabled,
+  buildPremiumLink
+} from 'cozy-client/dist/models/instance'
+
+import useInstance, { useInstanceState } from '../helpers/useInstance'
+import withLocales from '../I18n/withLocales'
+import fr from './locales/fr.json'
+import en from './locales/en.json'
+import { IllustrationDialog } from '../CozyDialogs'
+import { Spinner } from 'cozy-ui/react'
+import Icon from '../Icon'
+import CozyUpgradeIcon from '../Icons/CozyUpgrade'
+import Button from '../Buttons'
+import Typography from '../Typography'
+
+const locales = {
+  en,
+  fr
+}
+
+const OnlyOfficePaywall = ({ onClose, isPublic, t }) => {
+  const client = useClient()
+  const instance = useInstance(client)
+
+  const hasPremiumEnabled = arePremiumLinksEnabled(instance)
+
+  if (instance.state === useInstanceState.Idle) return null
+
+  if (
+    instance.state === useInstanceState.Loading &&
+    instance.state !== useInstanceState.Loaded
+  ) {
+    return (
+      <IllustrationDialog
+        size="small"
+        actionsLayout="column"
+        open={true}
+        content={
+          <div className="u-h-5">
+            <Spinner size="xxlarge" noMargin middle />
+          </div>
+        }
+        onClose={onClose}
+      />
+    )
+  }
+
+  const link = buildPremiumLink(instance)
+  let type = 'default'
+  if (hasPremiumEnabled && isPublic) {
+    type = 'public'
+  } else if (hasPremiumEnabled && link) {
+    type = 'premiumLink'
+  }
+
+  const content =
```

Comment:
on peut mettre ça dans le return directement et supprimer le `? :`, genre
``` 
content={
type === 'default' && <ReactMarkdown />
type !== 'default' &&  t(`onlyOfficePaywall.${type}.content`)
}
```

Reply:
I don't see how it's improve readability because you will need an React.frament. I remove the const and directly return the ternary expression